src/kernels/composable_kernel/include/kernel_algorithm/gridwise_generic_2d_reduction_blockwise.hpp

```diff
@@ -0,0 +1,462 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2020 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+#ifndef CK_GRIDWISE_GENERIC_2D_REDUCTION_BLOCKWISE_HPP
+#define CK_GRIDWISE_GENERIC_2D_REDUCTION_BLOCKWISE_HPP
+
+#include "float_type.hpp"
+#include "reduction_operator.hpp"
+#include "reduction_functions.hpp"
+#include "reduction_common.hpp"
+
+#include "blockwise_generic_tensor_slice_copy.hpp"
+#include "ConstantMatrixDescriptor.hpp"
+
+namespace ck {
+
+template <int BlockSize,
+          typename srcDataType,
+          typename dstDataType,
+          typename src2dDesc,
+          typename dst1dDesc,
+          typename compType,
+          ckReduceTensorOp_t op,
+          ckNanPropagation_t nanPropaOpt,
+          ckReduceTensorIndices_t reduceIndicesOpt,
+          int callId,
+          int GredAccessesPerThreadInBlock>
+struct Gridwise_generic_reduction_xy_to_x_blockwise
```

Comment:
Can you rename it to ```GridwiseReduction_xy_to_x_blockwise``` ? That is to remove ```generic``` keyword, and use CamelCase for struct name. ```_xy_to_x_blockwise``` is OK because you are trying to describe some implementation detail, and it would be hard to read if using camel case.

Pretty much every tensor is now a generic tensor in this code, so generic doesn't have much meaning now, we should have removed every generic keyword, but it's still in many places because of historic reason

Reply:
Good explanation !